api-client/src/main/java/com/takipi/api/client/request/alertssettings/UpdateAlertsSettingsRequest.java

```diff
@@ -0,0 +1,313 @@
+package com.takipi.api.client.request.alertssettings;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings.AlertsDefaultEmailSettings;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings.AlertsDefaultHipChatSettings;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings.AlertsDefaultPagerDutySettings;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings.AlertsDefaultSlackSettings;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings.AlertsDefaultServiceNowSettings;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings.AlertsDefaultWebhookSettings;
+import com.takipi.api.client.data.alertssetings.DefaultAlertsSettings.InitializedFields;
+import com.takipi.api.client.request.ServiceRequest;
+import com.takipi.api.client.result.EmptyResult;
+import com.takipi.api.core.request.intf.ApiPostRequest;
+import com.takipi.common.util.JsonUtil;
+
+import java.util.Map;
+
+public class UpdateAlertsSettingsRequest extends ServiceRequest implements ApiPostRequest<EmptyResult>
+{
+	private final DefaultAlertsSettings defaultAlertsSettings;

```

Comment:
This is an `ApiResult` class, and should be used as such.
This class can explicitly hold the actual settings classes as members.